### PR TITLE
use setFactory service definition method for symfony >= 2.6 (when possible)

### PR DESCRIPTION
**Pull request by adam187**
setFactoryService, setFactoryMethod is deprecated since 2.6

### CODE REVIEW

**Comment by digitalkaoz** [comment]
+1

**Comment by Soullivaneuh** [comment]
:+1: if this permit to remove the two page of deprecated warning on each composer install since symfony 2.7. :)

File: DependencyInjection/FOSUserExtension.php [hunk 1]
```diff
@@ -33,6 +34,18 @@ public function load(array $configs, ContainerBuilder $container)
             $container->setParameter($this->getAlias() . '.backend_type_' . $config['db_driver'], true);
         }
 
+        if ('custom' !== $config['db_driver'] && 'propel' !== $config['db_driver']) {
+            $managerService = ('orm' === $config['db_driver']) ? 'fos_user.entity_manager' : 'fos_user.document_manager';
+            $doctrineService = ('orm' === $config['db_driver']) ? 'doctrine' : sprintf('doctrine_%s', $config['db_driver']);
```
Comment (stof) [inline DependencyInjection/FOSUserExtension.php]: I would do the orm check only once.

**Comment by stof** [comment]
@adam187 can you update your PR ?

**Comment by adam187** [comment]
@stof you mean something like this ?

**Comment by stof** [comment]
@adam187 yes

**Comment by stof** [comment]
Thank you @adam187.

**Comment by Soullivaneuh** [comment]
@stof Any chance to cherry pick this for 1.3 branch? Thanks.

**Pull request closed without merging**